orchagent/lagids.lua

```diff
@@ -37,24 +37,41 @@ if op == "add" then
         end
         -- proposed lag id is different than that in database OR
         -- the portchannel does not exist in the database
-        -- If proposed lagid is available, return the same proposed lag id
-        if redis.call("sismember", "SYSTEM_LAG_ID_SET", tostring(plagid)) == 0 then
-            redis.call("sadd", "SYSTEM_LAG_ID_SET", tostring(plagid))
-            redis.call("srem", "SYSTEM_LAG_ID_SET", tostring(dblagid))
-            redis.call("hset", "SYSTEM_LAG_ID_TABLE", pcname, tostring(plagid))
-            return plagid
+        -- If proposed lagid is not available, lpop the first availabe ID
+        local index = redis.call("lpos", "SYSTEM_LAG_IDS_FREE_LIST", tostring(plagid))
+        if index == false then
+            local lagid = redis.call("lpop", "SYSTEM_LAG_IDS_FREE_LIST")
+            redis.call("hset", "SYSTEM_LAG_ID_TABLE", pcname, lagid)
+            redis.call("sadd", "SYSTEM_LAG_ID_SET", lagid)
+            if dblagid then
```

Comment:
dblgid should be a number here by above dblagid = tonumber(dblagid). Will this "if dblagid then" always be true for an integer here? Same concern for line 57 and line 69.